src/scripts/main.js

```diff
@@ -1 +1,74 @@
 'use strict';
+
+const promise1 = new Promise((resolve, reject) => {
+  document.addEventListener('click', () => {
+    resolve('First promise was resolved');
+  });
+
+  setTimeout(() => {
+    reject(new Error('First promise was rejected'));
+  }, 3000);
+});
+
+const promise2 = new Promise((resolve, reject) => {
+  document.addEventListener('click', () => {
+    resolve('Second promise was resolved');
+  });
+
+  document.addEventListener('contextmenu', (e) => {
+    e.preventDefault();
+    resolve('Second promise was resolved');
+  });
+});
+
+const promise3 = new Promise((resolve, reject) => {
+  let left = false;
+  let right = false;
+
+  document.addEventListener('click', () => {
+    left = true;
+
+    if (left === true && right === true) {
+      resolve('Third promise was resolved');
+    };
+  });
+
+  document.addEventListener('contextmenu', (e) => {
+    e.preventDefault();
+    right = true;
+
+    if (left === true && right === true) {
+      resolve('Third promise was resolved');
+    };
```

Comment:
It can be simplify with 1 listener [`mousedown ` event](https://developer.mozilla.org/en-US/docs/Web/API/Element/mousedown_event) then check `event.button`